lib/libc/minimal/include/sys/types.h

```diff
@@ -15,7 +15,9 @@ typedef unsigned int mode_t;
 #if !defined(__ssize_t_defined)
 #define __ssize_t_defined
 
+#define unsigned signed
```

Comment:
what is the root cause which makes this necessary?
surely there is a better way to fix this

Reply:
The `include/toolchain/xcc_missing_defs.h` has this comment
```
/*
 * xcc already defines __SIZE_TYPE__ as "unsigned int" but there is no way
 * to safeguard that here with preprocessor equality.
 */
```

Reply:
The commit 2a63e34 says
```
types.h was wrongly defining unsigned as signed and following
undefining it. This definition was not being used anywhere though.
```
so it seems that the change was actually not fixing anything. Thus the simplest solution is just to revert that commit and figure out other solution if that is really needed.

Reply:
The original commit that introduced the define/undef magic is this one 458e2ed1330d7e08c56bf5073ae3feadcfe8059d

Reply:
Indeed that commit was not fixing the problem. One to define ssize_t is based on SIZE_MAX that is part of C standard. ssize_t is just part of POSIX standard.

```
#if __SIZE_MAX__ == __UINT32_MAX__
#define SSIZE_MAX       __INT_MAX__
#elif __SIZE_MAX__ == __UINT64_MAX__
#define SSIZE_MAX       __INT64_MAX__
#else
#error "unexpected __SIZE_MAX__ value"
#endif
```

That' being said it seems that this can have some caveats that is not clear to me.
The problem with the undefine trick is that it *may* (not sure if the standard cover only macro or all identifier) violates C standard 
```
If the program removes (with #undef) any macro definition of an identifier in the first
group listed above, the behavior is undefined.
```
But it is violating other code guidelines (aka MISRA) for sure.